linux/kernel/building.md

```diff
@@ -109,18 +109,11 @@ If you find you need other things, please submit a pull request to change the do
 
 ### Install toolchain
 
-Use the following command to download the toolchain to the home folder:
-
+Use the following command to install the cross-arch compiler onto your machine:
 ```bash
-git clone https://github.com/raspberrypi/tools ~/tools
+sudo apt install binutils-arm-linux-gnueabihf crossbuild-essential-armhf
 ```
 
-Updating the $PATH environment variable makes the system aware of file locations needed for cross-compilation. 
-
-```bash
-echo PATH=\$PATH:~/tools/arm-bcm2708/arm-linux-gnueabihf/bin >> ~/.bashrc
-source ~/.bashrc
-```
 If you are using a 32-bit operating system (for example, our Raspberry Pi Desktop for PC), then you may need to install an additional set of libraries:
```

Comment:
Are the above-mentioned cross-compilers available in "Raspberry Pi Desktop for PC" ? If they are, does that mean the additional `zlib1g-dev:amd64` mentioned below isn't needed?

Reply:
It appears that the compilers are in the 32bit repo's as seen here https://packages.debian.org/stretch/i386/gcc-arm-linux-gnueabihf/download and after spinning up a vm the tools are there when needed. I am not familiar with zlib1g so I will leave that up to @native-api as he appears to be knowledgeable.

Reply:
So i took another look at the dependencies and it appears that zlib1g is required when pulling the cross compiler as seen [here](https://packages.debian.org/stretch/gcc-6-arm-linux-gnueabihf).

My only concern is that when installing on a 32bit platform it will install the 32bit version of zib1g so for some unknown reason that the amd64 version is required i think that could be a bug with the build system? i am grasping at straws at this point.

Reply:
Thanks for putting so much faith in me but I've never compiled the Pi kernel with 32-bit tools :-)
All my knowledge comes from diagnosing the problem that I've linked to in my PR.

Reply:
IIRC the `zlib1g-dev:amd64` was only needed on the "32-bit Raspberry Pi Desktop for PC" because the compiler-tools being installed from the tools-repo were themselves actually 64-bit (and so needed the 64-bit zlib). See #1413

But as the compiler-tools (installed from the repo) will now be 32-bit, then the compilers will only need the 32-bit zlib.

At least, that's the way I understand it, but @pelwell and @XECDesign probably understand it better than I do!

Reply:
Thank you for the clarification! That makes the most sense!